tests/test_main.py

```diff
@@ -0,0 +1,51 @@
+import pydantic
+import pytest
+import xarray as xr
+
+import xcollection
+
+ds = xr.tutorial.open_dataset('rasm')
+
+
+@pytest.mark.parametrize('datasets', [None, {'a': ds, 'b': ds}])
+def test_init(datasets):
+    c = xcollection.Collection(datasets)
+    assert isinstance(c.datasets, dict)
+    if datasets is not None:
+        assert len(c) == len(datasets)
+        assert set(c.keys()) == set(datasets.keys())
+
+
+def test_repr():
+    c = xcollection.Collection({'bar': ds, 'foo': ds})
+    assert 'foo' in repr(c)
+    assert 'bar' in repr(c)
+
+
+@pytest.mark.parametrize('datasets', [{'a': ds, 'b': 5}, {1: ds}, {'test': ds.Tair}])
```

Comment:
Not needed now but we *could* promote DataArrays to Datasets use `.to_dataset()`

Reply:
:+1:

Supporting this requires a small change and it would be convenient for users. I'm going to include this shortly